File: services/forum-service/src/models/Notification.js
const crypto = require('crypto');
const pool = require('../config/database');

class Notification {
  static async create({ recipient_aid, type, title, content = null, link = null, metadata = {} }) {
    const notification_id = `notif_${crypto.randomUUID()}`;
    const query = `
      INSERT INTO notifications (
        notification_id, recipient_aid, type, title, content, link, metadata, created_at
      )
      VALUES ($1, $2, $3, $4, $5, $6, $7::jsonb, NOW())
      RETURNING *
    `;

    const result = await pool.query(query, [
      notification_id,
      recipient_aid,
      type,
      title,
      content,
      link,
      JSON.stringify(metadata ?? {}),
    ]);

    return result.rows[0];
  }
}

module.exports = Notification;

File: services/identity-service/internal/models/notification.go
package models

import "time"

type Notification struct {
	ID             int64     `json:"id"`
	NotificationID string    `json:"notification_id"`
	RecipientAID   string    `json:"recipient_aid"`
	Type           string    `json:"type"`
	Title          string    `json:"title"`
	Content        string    `json:"content"`
	Link           string    `json:"link"`
	IsRead         bool      `json:"is_read"`
	Metadata       string    `json:"metadata"`
	CreatedAt      time.Time `json:"created_at"`
}
